ase/scripts/generate_ase_environment.py

```diff
@@ -393,39 +393,41 @@ def config_qsys_sources(filelist, vlog_srcs, vhdl_srcs):
     if (not qsys_srcs):
         return dict()
 
-    # First step: copy the trees holding Qsys sources to a temporary tree
+    # First step: copy the trees holding Qsys, IP sources to a temporary tree
     # inside the simulator environment.  We do this to avoid polluting the
     # source tree with Qsys-generated files.
-    copied_qsys_dirs = dict()
+    copied_qsys_ip_dirs = dict()
     copied_tcl_dirs = dict()
     tgt_idx = 0
     os.mkdir('qsys_sim')
-    qsys_srcs_copy = []
+    qsys_ip_srcs_copy = []
     tcl_srcs_copy = []
-    for q in qsys_srcs:
+    for q in srcs:
```

Comment:
Instead of switching to "srcs" here, why not change line 380 to look for both .qsys and .ip, putting both on the qsys_srcs list? I think a lot of the other changes you've had to make would then go away.

The current change would still fail if only .ip files are present and no .qsys. Line 394 would cause a return without expanding .ip files.

Reply:
I made that update , but I think I would still like to add ip in the variable names to reflect its working on qsys as well as ip files listed in sources.txt

Reply:
also tested it on p4 project provided by Ryan